### PR TITLE
Recycler

### PR DESCRIPTION
**Pull request by Lefunny**
https://user-images.githubusercontent.com/89319185/222737384-4f262e78-2768-4ba7-bf56-8e6d9659db5c.mp4

## About The Pull Request

New skill that recycles dead xenomorphs for psy points and larva points, the amount is entirely based on the caste and goes from 2 larva points as a tier one  to 8 larva points((an entire larva)) at tier 4.
## Why It's Good For The Game

Dying as xenomorph always gave an fair more opressive feeling than dying as marine. Not only you lost time to age depending on your caste but you also gave a significant amount of points. The new skill recycle aims to give xenos the possibility to at least fight for the bodies of fallen xenos and try to recover a bit from the lost. Also this skill is a hivelord exclusive making the caste more wanted and interesting.
## Changelog
:cl:
add: New hivelord ability recycle. Use it on dead xenos to gain some psy points and larva progress.
soundadd: recycle has a fun eating sound.
imageadd: recycle ability sprite.

/:cl:

### CODE REVIEW

**Comment by Lefunny** [comment]
i have yet to work in a new skill icon for it and also a new sound. Numbers can always be rebalanced

**Comment by FosterGrime-Heymynameslouie** [comment]
isint this just the salvage ability from way back then?

if so very cool I kinda missed that aiblity

**Comment by Naaanii** [comment]
Rewards for dying is pretty bad/awkward design.

**Comment by MiniMeatwad** [comment]
Removing the corpse outright is a bity iffy, because it prevents marines from being able to get req points from the corpse. As an alternative, you could make it get teleported to a resin silo, so xenos have to hide corpses somewhere and marines can still get req points for map control.

As for the larva point reward being a bit high and leaning into a sort of suicide as xeno playstyle, you could only give 1-2 larva points per corpse.

Could also do 8 for dead larvae so xenos can pull shenanigans, the first free xeno death mechanic of its kind (fair an balanse). Make it into a reward to recover the corpses so they aren't turned into req points, but don't make it too good, most of the reward should come from denying req points from marines anyways.

**Comment by RaidarV** [comment]
There is no way this will drastically change how sweaty xenos play the game and make the gameplay more obnoxious. No way this can go wrong.

That aside, what kinda cooldown do you think about putting this on ?

**Comment by Al-1ce** [comment]
I can swear the ability to do this was a thing like 3 years ago.

**Comment by Goblinu** [comment]
Dont you think "reabsorb" is a more fitting name for the skill rather than recycle? Recycle makes it sound like xenos are tin cans needing to be picked up and recycled

**Comment by Oscari-pog** [comment]
> Rewards for dying is pretty bad/awkward design.

It is a reward for securing the dead xeno body instead of letting Marines take it. It is still a net loss.

**Comment by Lefunny** [comment]
> Rewards for dying is pretty bad/awkward design.

Honestly i can't see any scenario were xenos leave in the positive side if they wanted to somehow use their own deaths. Psy points wise we go from 20-40, 20 being a tier one and 40 a t4, even the death of a shrike would be still a really bad thing even if xenos could get a few psypoints out of it. The larva points can be indeed changed but i am not sure yet if it is this strong. Four t1s have to die for one larva, three t2s or two t3s. the t3s are fair more of a soft situation but four t1s dying is basically a xeno wipe, three t2s dying is a massive loss. 

In resume i do not think there is any reward for dying there, you just lose a bit less than the usual and deny points to req. Of course if you win the fight that will happen for that sweet sweet queen body.

**Comment by Lefunny** [comment]
> There is no way this will drastically change how sweaty xenos play the game and make the gameplay more obnoxious. No way this can go wrong.
> 
> That aside, what kinda cooldown do you think about putting this on ?

Cooldown if you mean after using the skill probably none. It is probably already clear i want to make Recycle the counterpart of defib so just like the tool the only thing required for it to work is the xeno standing in place for seven seconds without taking any damage.

**Comment by iforgotmeword** [comment]
> 3. Dead larvae have two problems, the first being damn massive skill issue if you die as larva, second is i think the larva bodie that spawns after surgery is the same of an actual dead larva. It would make xenos able to find this larva stacks and recycle for massive amounts of xenos without any loss.

Yea I don’t think the surgery larva is the same as I unable to sell them unlike the ones we kill 
They count as T1 xenos if gutted and sold they also act as body’s not items if killed
So i don’t think Xenos will rush with larva as it still giving marines points

**Comment by XElectricX** [comment]
Give it to Drones as well, this was their ability.

**Comment by MiniMeatwad** [comment]
Although I don't like this change, if merged it does change how the game works at large.

Reverse the order if nothing else, and consider reducing it by half later. Since all you're doing is lessening the penalty of death, give T1s more larva return so you can suicide as runner for the 6th time and not cause xenos to lose on that alone, but if you want the firepower of T3, you pay for it when you die.

Larva = 8 points
T4 = 0 points, maybe don't die as Queen?

**Comment by Lefunny** [comment]
> Although I don't like this change, if merged it does change how the game works at large.
> 
> Reverse the order if nothing else, and consider reducing it by half later. Since all you're doing is lessening the penalty of death, give T1s more larva return so you can suicide as runner for the 6th time and not cause xenos to lose on that alone, but if you want the firepower of T3, you pay for it when you die.
> 
> And consider Larva = 8 points, but marines can gib larva corpses with a knife, as it's an in-game remedy to xenos accidentally killing themself as a larva and having to adminhelp it. If that's outside your scope, do as you please.

About changing the xeno order i will wait for the pr to be merged in order to start any rebalance. I find it genuinely hard to rebalance numbers with only assumptions about their effects in a normal round. For the larvas it surely can be done but marines being able to gib larvas is not necessary. Not only larvas can be sold in req they are gibbed in 90% of the cases because of their really low health.

**Comment by Lefunny** [comment]
> Give it to Drones as well, this was their ability.

No, this skill gives psy points and larva points. With the second main ping being giving hivelord some more spotlight. Giving it to drone ((at least in the state of the skill rn)) would break the second point with people playing drone in a pure combat way fighting for bodies themselves.

**Comment by Autisem** [comment]
I like this change because its something equivelent to marines reviving.

A marine can be revived from death due to there own mistakes very easily with NO COST to them, dying doesnt matter aslong as you get revived(there are injuries but thats minor) compared to xenos who are just dead dead.

Letting xenos recoup losses is nice way to avoid marines just stomping them if they fuck up once

**Comment by zzurge** [comment]
it worked back in the day so i can see it working today,Gonna have to emphasize that someone bring a powerfist or bike to drag corpses now.maybe people will stop forgetting to setup the research lab for money

**Comment by Lumipharon** [comment]
Being able to dispose of bodies is neat, but I'm somewhat dubious about the level of larval recycling you're getting, considering how larva rate mechanics work.

**Comment by chizzy376** [comment]
> Also that sound needs to die.

We already ripped our cocooning sounds from virgo, I don't see why this is bad.

File: code/modules/mob/living/carbon/xenomorph/castes/hivelord/castedatum_hivelord.dm [hunk 1]
```diff
@@ -59,6 +59,7 @@
 		/datum/action/xeno_action/watch_xeno,
 		/datum/action/xeno_action/activable/psydrain,
 		/datum/action/xeno_action/activable/cocoon,
+		/datum/action/xeno_action/activable/Recycle,
```
Comment (novaepee) [inline code/modules/mob/living/carbon/xenomorph/castes/hivelord/castedatum_hivelord.dm:62]: remove the captial R, keep the standard for xeno abilities to not have captial letters

**Pull request closed without merging**